bolt-aws-lambda/src/main/java/com/slack/api/bolt/aws_lambda/SlackApiLambdaHandler.java

```diff
@@ -98,4 +87,33 @@ public ApiGatewayResponse handleRequest(ApiGatewayRequest input, Context context
         return results;
     }
 
-}
\ No newline at end of file
+    protected ApiGatewayResponse toApiGatewayResponse(Response slackResp) {
+        return ApiGatewayResponse.builder()
+                .statusCode(slackResp.getStatusCode())
+                .headers(toStringToStringMap(slackResp.getHeaders()))
+                .rawBody(slackResp.getBody())
+                .build();
+    }
+
+    private static Map<String, String> toStringToStringMap(Map<String, List<String>> stringToStringListMap) {
+        Map<String, String> headers = new HashMap<>();
+        for (Map.Entry<String, List<String>> each : stringToStringListMap.entrySet()) {
+            if (each.getValue() != null && !each.getValue().isEmpty()) {
+                headers.put(each.getKey(), each.getValue().get(0)); // set the first value in the array
+            }
+        }
+        return headers;
+    }
+
+    /**
+     * Allows access to the internal {@link App} used by this class interact
+     * with Slack. It is useful in cases where changes on the app are necessary
+     * after initialisation. For example, when adding a reaction after certain
+     * command calls this bot.
+     *
+     * @return the {@code App} used by this class to handle events
+     */
+    protected App app() {
```

Comment:
I was thinking `app()` may be also fine here. I'm fine with `app()` instead of `getApp()`.